### ISSUE TITLE
[FEATURE_REQUEST] Widget to show qBittorrent queue information

### ISSUE THREAD

**Issue by Scootter23** [Lissy93/dashy#1122]
<h3 dir="auto">Is your feature request related to a problem? If so, please describe.</h3>
<p dir="auto">Like the SABnzbd widget it would be great to have a widget to show qBittorrent queue information and or stats and monitoring.</p>
<h3 dir="auto">Describe the solution you'd like</h3>
<p dir="auto">Most of the torrent applications share some of the same features and the widget could maybe work for multiple clients. If not one widget I would think once this is done it would not take much to make the others.</p>
<p dir="auto">qBitTorrent Official Website - <a href="https://www.qbittorrent.org/" rel="nofollow">https://www.qbittorrent.org/</a><br>
qBittorrent API documentation - <a href="https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)">https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)</a></p>
<p dir="auto">A couple screenshots from browser extensions that do a similar job. Just some ideas.<br>
<a target="_blank" rel="noopener noreferrer nofollow" href="https://user-images.githubusercontent.com/20101458/221382355-a74a8ab0-2f71-456f-814e-9ab9fbe62859.jpg"><img src="https://user-images.githubusercontent.com/20101458/221382355-a74a8ab0-2f71-456f-814e-9ab9fbe62859.jpg" alt="torrent-extension2" style="max-width: 100%;"></a><br>
<a href="https://chrome.google.com/webstore/detail/remote-deluge/epcolmlaegimilnokaddncgmdafacohf?hl=en" rel="nofollow">https://chrome.google.com/webstore/detail/remote-deluge/epcolmlaegimilnokaddncgmdafacohf?hl=en</a></p>
<p dir="auto"><a target="_blank" rel="noopener noreferrer nofollow" href="https://user-images.githubusercontent.com/20101458/221382356-3b59d36f-ce4c-413a-b8e2-5ed493bb7d2a.jpg"><img src="https://user-images.githubusercontent.com/20101458/221382356-3b59d36f-ce4c-413a-b8e2-5ed493bb7d2a.jpg" alt="torrent-extension1" style="max-width: 100%;"></a><br>
<a href="https://chrome.google.com/webstore/detail/torrent-to-transmission/fjfeeonbeiocojnpfboldpckcgcfknll?hl=en" rel="nofollow">https://chrome.google.com/webstore/detail/torrent-to-transmission/fjfeeonbeiocojnpfboldpckcgcfknll?hl=en</a></p>
<h3 dir="auto">Priority</h3>
<p dir="auto">Low (Nice-to-have)</p>
<h3 dir="auto">Is this something you would be keen to implement</h3>
<p dir="auto">No</p>

<blockquote>                                <div>          <strong>                          <a href="https://www.qbittorrent.org/">qBittorrent Official Website</a>                      </strong>        </div>                    <div>qBittorrent Official Website</div>                      </blockquote><blockquote>                    <img src="https://opengraph.githubassets.com/b132f43fe1c6ad98d5864904d9b1a0209fc2b1980c99c1fe7c92da6f67005e85/qbittorrent/qBittorrent" width="48" align="right" />                          <div>          <strong>                          <a href="https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)">WebUI API (qBittorrent 4.1)</a>                      </strong>        </div>                    <div>qBittorrent BitTorrent client. Contribute to qbittorrent/qBittorrent development by creating an account on GitHub.</div>                      </blockquote><blockquote>                    <img src="https://lh3.googleusercontent.com/seMO7UfE5KG-0_yCnw4Q-YWIv4ih3HoFiR9jHM2R1D05lNL246XEQaV3sI4cXD7oSVdO14RbEmQFuD65bu_tT9LKrJ0&#x3D;w128-h128-e365-rj-sc0x00ffffff" width="48" align="right" />                          <div>          <strong>                          <a href="https://chrome.google.com/webstore/detail/remote-deluge/epcolmlaegimilnokaddncgmdafacohf">Remote Deluge</a>                      </strong>        </div>                    <div>Provides a quick overview of torrent status for Deluge.</div>                      </blockquote><blockquote>                    <img src="https://lh3.googleusercontent.com/BtwmbgDQR2hiZTLta063MFS347LSWC_w80mVaan2CMuc2-ywrLehcv642ty04MyWGlcSufvff7EXHe8aVqKSIVzG&#x3D;w128-h128-e365-rj-sc0x00ffffff" width="48" align="right" />                          <div>          <strong>                          <a href="https://chrome.google.com/webstore/detail/torrent-to-transmission/fjfeeonbeiocojnpfboldpckcgcfknll">.torrent to Transmission</a>                      </strong>        </div>                    <div>Automagically add, access and control torrents with Chrome.</div>                      </blockquote>

**Comment by Scootter23** [comment]
Still looking for something like this. I did some programming a long time ago and even though I am ancient I might try to see what I can come up with.

**Comment by Scootter23** [comment]
I was thinking once the Prometheus Data widget is done I could just use that for my qBittorrent data since I already pull that data with a Prometheus exporter.

**Comment by jpcapone** [comment]
As another non-coder, I would just like to second the desire to have this plugin.

**Comment by AutoGitr** [comment]
+1 for this as a qBittorrent widget. 

It may be worth looking at how [https://github.com/causefx/Organizr](url) does it. It has built in widgets for Radarr, Sonarr, Sabnzbd, qBittorrent, Tautulli, Plex and Overseerr which are all very popular and suitable for widgets. There is also a calendar widget that connects to Radarr and Sonarr which would be very useful in Dashy.

**Comment by githubaccount3322** [comment]
+1

**Comment by flayzer** [comment]
+1 for a qBittorrent (or other torrent downloader) widget.

I'm thoroughly enjoying Dashy and a qBittorrent widget in the same style as the sabnzbd widget would make it even better.

Thanks for all your hard work!

**Comment by averymd** [comment]
This would still be really awesome!
